packages/agreement-process/src/model/domain/validators.ts

```diff
@@ -452,25 +446,19 @@ export const filterVerifiedAttributes = (
       )
   ) as VerifiedTenantAttribute[];
 
-const notRevocatedTenantAttributesFilter = <
-  T extends RevocableTenantAttribute | NotRevocableTenantAttribute
->(
-  att: T[]
-): ((a: T) => boolean) =>
-  match(att)
-    .with(
-      P.array({ revocationTimestamp: P.instanceOf(Date) }),
-      () =>
-        (a: RevocableTenantAttribute): boolean =>
-          !a.revocationTimestamp
-    )
-    .otherwise(() => () => true);
-
-export const failOnActivationFailure = (
-  newState: AgreementState,
-  agreement: Agreement
-): void => {
-  if (agreementActivationFailureStates.includes(newState)) {
-    throw agreementActivationFailed(agreement.id);
-  }
-};
+const filterNotRevocatedAttributes = (
+  consumerAttributes: TenantAttribute[]
+): TenantAttribute[] =>
+  consumerAttributes.filter((att) =>
+    match(att)
+      .with(
+        { type: tenantAttributeType.CERTIFIED },
+        (a) => !a.revocationTimestamp
+      )
+      .with(
+        { type: tenantAttributeType.DECLARED },
+        (a) => !a.revocationTimestamp
+      )
+      .with({ type: tenantAttributeType.VERIFIED }, () => true)
```

Comment:
a verified attribute is valid for a given eservice if the producer of the eservice is in `verifiedBy` array and the verification is not expired (see [here](https://github.com/pagopa/interop-be-agreement-process/blob/1.0.x/lifecycle/src/main/scala/it/pagopa/interop/agreementprocess/lifecycle/AttributesRules.scala#L50))

FYI: it is similar to `revocationTimestamp`, but these attributes can be verified by more than one organization. 
`verifiedBy` array contains all organizations that have currently verified the attribute for the tenant.